src/BizHawk.Client.EmuHawk/tools/TAStudio/TAStudio.cs

```diff
@@ -554,6 +554,8 @@ private bool StartNewMovieWrapper(ITasMovie movie, bool isNew)
 
 			SuspendLayout();
 			WantsToControlStopMovie = false;
+			// Stop the movie here, so that when main form calls movie stop, it won't save. (we've already saved, if needed)
+			MovieSession.StopMovie(false);

```

Comment:
I don't really like this. The problem here is that you're throwing out the current movie, but `StartNewMovie` could still fail to load the new movie (via the active cheat check for example, or just straight up unexpected error on reboot or something), in which case you would want the current movie to remain.

Perhaps using `CurrentTasMovie.ClearChanges()` here is actually the cleanest method. That way the movie stays until the new is loaded but won't be saved.

Reply:
That's a good point. However, failure can still happen after MainForm's call to MovieSession.StopMovie so it would be unrecoverable anyway. MainForm stops the current movie on CloseGame (which happens during TAStudio loading of a movie), rather than wait until it is time to load the new one. (I'm not sure what the cause is, but I have had TAStudio and even EmuHawk crash/close themselves on movie load failures. I typically just re-open EmuHawk and try again and that usually works.)

I think the proper solution to that problem is not anything in TAStudio, but would require MovieSession.StopMovie to not throw away the movie, and make corresponding changes elsewhere. I think that's outside the scope of this PR.

Reply:
> Is there a problem with using CurrentTasMovie.ClearChanges() instead?

I don't think so. (putting it back in the ask save method; `CurrentTasMovie` might be `null` here because it could be a regular movie)

> If the logic in mainform was changed, this code would need to get changed too.

I don't see why. If stopping the movie doesn't throw it away, then it can be restarted without any fuss by calling `SwitchToPlay`. Although I suppose it may simplify the needed changes in MainForm if we don't stop the movie here.
So I think yeah it probably is better to use ClearChanges.